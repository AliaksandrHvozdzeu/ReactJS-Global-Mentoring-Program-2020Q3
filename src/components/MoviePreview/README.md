#Movie preview component

![image](https://drive.google.com/uc?export=view&id=1_RSb-zPJ_g8z-Ij_0Vs2qWSqegh8G4zy)


