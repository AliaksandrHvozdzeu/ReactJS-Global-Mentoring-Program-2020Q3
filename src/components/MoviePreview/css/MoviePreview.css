.close-preview {
    position: absolute;
    width: 1.5rem;
    height: 1.5rem;
    top: 0;
    right: 0;
    color: transparent;
    background-repeat: no-repeat;
    transform: rotate(90deg);
    overflow: hidden;
}

.preview-header-form {
    height: 50vh;
    background-image: url(/images/header.jpg);
    background-repeat: no-repeat;
    background-position: center;
    background-size: cover;
    position: relative;
    overflow: hidden;
}

.movie-preview {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: stretch;
    height: 105%;
    padding: 5vh 0;
    top: 40px;
    position: absolute;
    margin-left: 69px;
}

.unselectable {
    -webkit-touch-callout: none; /* iOS Safari */
    -webkit-user-select: none; /* Chrome/Safari/Opera */
    -khtml-user-select: none; /* Konqueror */
    -moz-user-select: none; /* Firefox */
    -ms-user-select: none; /* Internet Explorer/Edge */
    user-select: none;
    /* Non-prefixed version, currently
                           not supported by any browser */
}

.preview-wrapper {
    margin: 0 6%;
    position: relative;
    height: 100%;
    width: 88%;
    overflow: hidden;
}

.preview-header {
    height: 50vh;
    background-image: url(/images/header.jpg);
    background-repeat: no-repeat;
    background-position: center;
    background-size: cover;
    position: relative;
    overflow: hidden;
}

.close-preview-button {
    position: absolute;
    top: 50px;
    right: 2px;
    outline: none;
}

