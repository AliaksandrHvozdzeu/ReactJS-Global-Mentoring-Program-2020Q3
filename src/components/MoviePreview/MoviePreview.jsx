import React from 'react';
import PropTypes from 'prop-types';
import MovieDetails from '../MovieDetails';
import Brand from '../Brand';
import img from '../../../public/images/search_42.png';
import './style.css';

export default function MoviePreview({ details, closePreview }) {

  return (
    <>
      <div className="preview-wrapper">
        <Brand/>
        <div className="movie-preview unselectable">
          <MovieDetails details={details}/>
          <button type="button" onClick={closePreview} onKeyDown={closePreview}>
            <img src={img} className="close-preview" alt="close preview"/>
          </button>
        </div>
      </div>
    </>
  );

}

MoviePreview.propTypes = {
  details: PropTypes.func.isRequired,
  closePreview: PropTypes.func.isRequired,
};
