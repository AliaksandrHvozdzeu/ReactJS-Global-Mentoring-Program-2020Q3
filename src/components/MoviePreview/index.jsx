import React from 'react';
import Poster from '../MoviePoster';
import MovieDetails from '../MovieDetails';
import PropTypes from 'prop-types';
import './style.css';
import Brand from '../Brand';
import img from '../../../public/images/search_42.png';

const MoviePreview = ({ details, closePreview }) => (
    <div className="preview-wrapper">
      <Brand/>
      <div className="movie-preview unselectable">
        <Poster src={details.poster} alt={details.alt}/>
        <MovieDetails details={details}/>
        <img src={img} className="close-preview" alt="close preview" onClick={closePreview}/>
      </div>
    </div>
);

MoviePreview.propTypes = {
  details: PropTypes.func.isRequired,
  closePreview: PropTypes.func.isRequired,
};

export default MoviePreview;
