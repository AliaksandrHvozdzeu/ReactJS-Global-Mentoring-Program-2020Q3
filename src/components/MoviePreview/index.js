import MoviePreview from './js/MoviePreview';

export default MoviePreview;
