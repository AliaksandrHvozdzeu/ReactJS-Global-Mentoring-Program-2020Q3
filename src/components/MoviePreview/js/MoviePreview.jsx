import React from "react";
import PropTypes from "prop-types";
import { render } from "react-dom";
import { Provider } from "react-redux";
import MovieDetails from "../../MovieDetails";
import Brand from "../../Brand";
import "../css/MoviePreview.css";

import store from "../../../store";
import SearchPanel from "../../SearchPanel";

export default function MoviePreview({ details }) {

  const onPushHistoryState = () => {
    window.history.pushState(null, null, `/`);
  };

  const onClosePreview = () => {
    const header = document.getElementById("header");
    header.className = null;
    onPushHistoryState();
    render(<Provider store={store}><SearchPanel /></Provider>, header);
    window.history.pushState(null, null, `#/search?query=${localStorage.getItem("searchValue")}`);
  };

  return (
    <>
      <div className="preview-wrapper">
        <Brand />
        <div className="movie-preview unselectable">
          <MovieDetails details={details} />
        </div>
        <button className="close-preview-button"
                type="button"
                onClick={onClosePreview}
                onKeyDown={onClosePreview}>
          <img src="../images/search_42.png"
               className="close-preview"
               alt="close preview" />
        </button>
      </div>
    </>
  );

}

MoviePreview.propTypes = {
  details: PropTypes.shape({
    title: PropTypes.string,
    vote_average: PropTypes.string,
    genres: PropTypes.arrayOf(PropTypes.string),
    release_date: PropTypes.string,
    runtime: PropTypes.number,
    overview: PropTypes.string
  }).isRequired
};
