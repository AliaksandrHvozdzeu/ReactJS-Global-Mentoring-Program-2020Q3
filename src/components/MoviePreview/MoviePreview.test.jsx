import React from "react";
import rerender from "react-test-renderer";
import MoviePreview from "./js/MoviePreview";
import { mount, shallow } from "enzyme";
import { jest } from "@jest/globals";

describe("when MoviePreview", () => {

  const details = {
    "id": 399055,
    "title": "The Shape of Water",
    "tagline": "A Fairy Tale for Troubled Times",
    "vote_average": "7.3",
    "vote_count": 3200,
    "release_date": "2017-12-01",
    "poster_path": "https://image.tmdb.org/t/p/w500/k4FwHlMhuRR5BISY2Gm2QZHlH5Q.jpg",
    "overview": "An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.",
    "budget": 19500000,
    "revenue": 185545281,
    "genres": ["Drama", "Fantasy", "Romance"],
    "runtime": 123
  };

  test("then snapshot created", () => {
    const component = rerender.create(<MoviePreview details={details} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('1', () => {
    const component = shallow(<MoviePreview details={details} />);
    const onClosePreview = jest.fn();
    expect(
      component.find('.close-preview-button').contains([
        <button className="close-preview-button"
                type="button"
                onClick={onClosePreview}
                onKeyDown={onClosePreview}>
          <img src="../public/images/search_42.png"
               className="close-preview"
               alt="close preview" />
        </button>
      ]),
    );
  });

  test('onClosePreview ', () => {
    const wrapper = mount(<MoviePreview details={details} />);
    expect(wrapper.find('.preview-wrapper')).toBeTruthy();
    const onClosePreview = jest.fn();
    expect(onClosePreview).toBeTruthy();
  });

});
