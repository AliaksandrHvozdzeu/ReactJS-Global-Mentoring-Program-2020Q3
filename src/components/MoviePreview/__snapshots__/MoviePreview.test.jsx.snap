// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when MoviePreview then snapshot created 1`] = `
<div
  className="preview-wrapper"
>
  <div
    className="brand"
  >
    <p
      className="brand-title"
    >
      <a
        className="brand-title-name"
        href=" #"
        id="blink"
        onClick={[Function]}
      >
        netflix
      </a>
      roulette
    </p>
  </div>
  <div
    className="movie-preview unselectable"
  >
    <img
      alt="Movie poster"
      className="poster"
      onClick={[Function]}
      onError={[Function]}
      role="presentation"
      src="https://image.tmdb.org/t/p/w500/k4FwHlMhuRR5BISY2Gm2QZHlH5Q.jpg"
    />
    <div
      className="movie-preview-description-block"
    >
      <h1
        className="movie-preview-title"
      >
        The Shape of Water
      </h1>
      <p
        className="movie-preview-rating"
      >
        7.3
      </p>
      <p
        className="movie-preview-genres"
      />
      <p
        className="movie-preview-release"
      >
        2017
      </p>
      <p
        className="movie-preview-runtime"
      >
        123
      </p>
      <p
        className="movie-preview-description"
      >
        An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.
      </p>
    </div>
  </div>
  <button
    className="close-preview-button"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    <img
      alt="close preview"
      className="close-preview"
      src="../images/search_42.png"
    />
  </button>
</div>
`;
